### PR TITLE
salt: workaround beacon.present bug

### PR DESCRIPTION
**Pull request by alexandre-allard**
**Component**: salt

**Context**: 
  - https://github.com/saltstack/salt/issues/58006
  - https://github.com/saltstack/salt/issues/58579
  - https://github.com/saltstack/salt/issues/59075
  - https://github.com/saltstack/salt/pull/58854

**Summary**:
Do not rely on `- save: True` and write the beacons configuration file using `file.managed`.

**Acceptance criteria**:

### CODE REVIEW

File: salt/metalk8s/beacon/certificates.sls [hunk 1]
```diff
@@ -39,15 +38,29 @@ Add beacon for certificates expiration:
...
     - interval: {{ certificates.beacon.interval }}
     - disable_during_state_run: True
     - notify_days: {{ certificates.beacon.notify_days }}
     - files: {{ kubeconfigs | json }}
 {%- endif %}
 
-{%- if not certs and not kubeconfigs %}
+{%- if certs or kubeconfigs %}
+{# We should use `- save: True` in the `beacon.present` states above,
+   instead of writting the configuration file with a `file.managed`,
```
Comment (gdemonet) [inline salt/metalk8s/beacon/certificates.sls]:
```suggestion
   instead of writing the configuration file with a `file.managed`,
```

**Review by gdemonet — DISMISSED** [review]
OK LGTM, feel free to ignore the typo if you get green builds :wink:

File: salt/metalk8s/beacon/certificates.sls (hunk 1 above)
Comment (alexandre-allard) [reply salt/metalk8s/beacon/certificates.sls]: Mêh, build failed... at least this is now fixed...

**Review by gdemonet — DISMISSED** [review]

**Review by thomasdanan — DISMISSED** [review]
After upgrading to the content of this PR, the beacon error is no present anymore in journal logs

**Review by NicolasT — DISMISSED** [review]

**Review by gdemonet — APPROVED** [review]

**Review by TeddyAndrieux — COMMENTED** [review]
~Not directly link to this PR but for me "beacon.certificate" should restart the salt-minion process. otherwise, it's not automatically taken into account during upgrade, especially since this state come after the "salt-minion reconfiguration"~

Edit: Hmm beacon are present in memory so ok not really needed

**Review by TeddyAndrieux — APPROVED** [review]
Tested it works for me !

**Pull request merged**